repository_service_tuf/cli/admin/helpers.py

```diff
@@ -280,6 +349,31 @@ def _load_key_from_file_prompt() -> SSlibKey:
     return key
 
 
+def _load_key_from_hsm_prompt() -> Optional[Key]:
+    """Prompt for a public key."""
+
+    def file_filter(f):  # pragma: no cover
+        return os.path.isdir(f) or os.path.isfile(f)
+
+    message = prompt_toolkit.HTML(
+        "Enter file path to a <b>CERTIFICATE</b> PEM key: "
+    )
+    path = _prompt_key(message, file_filter)
+
+    with open(path, "rb") as f:
+        cert_pem = f.read()
+
+    cert = load_pem_x509_certificate(cert_pem)
+
+    key = SSlibKey.from_crypto(cert.public_key())
+    key.unrecognized_fields[KEY_NAME_FIELD] = (
+        cert.subject.get_attributes_for_oid(oid.NameOID.COMMON_NAME)[0].value
```

Comment:
I get an error here. Will DM stack trace.

Reply:
Hi @trishankatdatadog, thank you!
I fixed it in the fixup commit, I was assuming that all keys have it.
Please, take a look at my proposal. Another option is we don't use it at all.